rhino/src/main/java/org/mozilla/javascript/optimizer/BodyCodegen.java

```diff
@@ -3728,6 +3734,84 @@ private void visitIncDec(Node node) {
         }
     }
 
+    // Handles super.x++ and variants thereof. We don't want to create new icode in the interpreter
+    // for this edge case, so we will transform this into something like super.x = super.x + 1
+    private void visitSuperIncDec(Node node, Node child, int incrDecrMask) {
+        Node object = child.getFirstChild();
+
+        // Push the old value on the stack
+        generateExpression(object, node); // [super]
+        cfw.add(ByteCode.DUP); // [super, super]
+        switch (child.getType()) {
+            case Token.GETPROP:
+                cfw.addALoad(contextLocal);
+                cfw.addALoad(variableObjectLocal);
+                cfw.addALoad(thisObjLocal);
+                cfw.addLoadConstant(0); // no_warn flag
+                addDynamicInvoke(
+                        "PROP:GETSUPER:" + child.getFirstChild().getNext().getString(),
+                        Signatures.PROP_GET_SUPER);
+                break;
+
+            case Token.GETELEM:
+                generateExpression(object.getNext(), node);
+                cfw.addALoad(contextLocal);
+                cfw.addALoad(variableObjectLocal);
+                cfw.addALoad(thisObjLocal);
+                addDynamicInvoke("PROP:GETELEMENTSUPER", Signatures.PROP_GET_ELEMENT_SUPER);
+                break;
+
+            default:
+                Codegen.badTree();
+        }
+
+        // stack: [super, p]
+        if ((incrDecrMask & Node.POST_FLAG) != 0) {
+            // For postfix we want a copy of the old value on the stack
+            cfw.add(ByteCode.SWAP); // [p, super]
+            cfw.add(ByteCode.DUP2); // [p, super, p, super]
+            cfw.add(ByteCode.POP); // [p, super, p]
+        }
+
+        // Increment or decrement the value
+        addObjectToDouble(); // Unbox
```

Comment:
I'm not sure that this is more efficient than leaving the object as it was, and adding a "MATH:ADD"  instruction, or a call to ScriptRuntime.subtract, instead.

Reply:
We've copy&pasted the code from the pre-existing `visitIncDec`. If we want to change the approach, it would be better to do it in both places (so, IMHO in a different PR)

Reply:
I did some experimenting with increment stuff last week and it's both not simple to completely change this stuff, and not faster to do it the way that I suggested. So I think that the way you have it here is fine. Thanks!